src/test/crimson/seastore/test_transaction_manager.cc

```diff
@@ -1247,6 +1310,59 @@ struct transaction_manager_test_t :
     });
   }
 
+  void test_clone_and_remap_pin() {
```

Comment:
I need to make sure whether this is a valid use case from ObjectStore perspective -- data overwrite and clone happening in the same transaction.

If yes, how about the nested case e.g. remap-clone-remap-... in the same transaction? And things are more complicated with the delta-base overwrite support (see https://github.com/ceph/ceph/pull/50419 @myoungwon ), we will need to consider to test interleaving that with clone&remap in various ways.

If no, things will be simpler to assert that the cloned extent must be stable. And I think the more important test would be to interleave various clone operations as independent transactions into `test_random_writes_concurrent()`.

@athanatos what's your insight?

Reply:
Looks this is testing clone-then-remap in the same transaction.

Probable we can also assert that the cloning extent is stable to detect whether the scenario is expected ?

Reply:
> If no, things will be simpler to assert that the cloned extent must be stable. And I think the more important test would be to interleave various clone operations as independent transactions into `test_random_writes_concurrent()`.

I think you mean `test_overwrite_pin()`, right? Because `test_random_writes_concurrent()` is modifying extents through the "mutate" interface, which is not supported by ObjectDataBlock yet. At the present, cloning is completely for ObjectDataBlock and hasn't taken mutations into account.

Reply:
> Probable we can also assert that the cloning extent is stable to detect whether the scenario is expected ?

Actually, CLONE only operates on lba mappings and doesn't see those extents being cloned.

Reply:
> And I think the more important test would be to interleave various clone operations as independent transactions into `test_random_writes_concurrent()`.

`test_random_writes_concurrent()` modify TestBlocks through extents' mutate interface, which the current ObjectDataBlock and CLONE don't support. Maybe we can implement these in follow-up PRs.

Reply:
> Actually, CLONE only operates on lba mappings and doesn't see those extents being cloned.

CLONE will always query the direct mapping which will get access to the cached extent if present in cache, so a check that the cloned extent should not be pending is possible, right?

Reply:
> test_random_writes_concurrent() modify TestBlocks through extents' mutate interface, which the current ObjectDataBlock and CLONE don't support. Maybe we can implement these in follow-up PRs.

Not sure how TestBlock's mutate interface is related here as we will only test to clone "stable" extents for now. But it should be good to implement in another PR.